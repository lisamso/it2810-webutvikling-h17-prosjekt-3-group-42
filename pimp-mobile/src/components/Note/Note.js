import React, { Component } from 'react';
import {Text, View} from "react-native";
import PropTypes from 'prop-types';
import { styles } from '../../styles.js';

class Note extends Component {

  render() {

    return(
      <View style={styles.note}>
        <Text style={[styles.title, styles.carrot]}>{this.props.title}</Text>
        <Text style={[styles.text, styles.carrot]}>{this.props.text}</Text>
      </View>
    )
  }
}

Note.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
}

export default Note