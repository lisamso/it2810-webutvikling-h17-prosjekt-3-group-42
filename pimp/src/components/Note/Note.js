import React, { Component } from 'react';
import PropTypes from 'prop-types'; // ES6
import {
  removeId,
  timestampAt
} from '../../helper';

class Note extends Component {
  constructor(props) {
    super(props)
    this.state = {
      title: props.title,
      text: props.text,
      titleFocused: false,
      textFocused: false,
      lastChanged: props.lastChanged,
      timestamp: null,
      saved: false,
    }
    this.handleTitleChange = this.handleTitleChange.bind(this)
    this.handleTextChange = this.handleTextChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  componentDidMount() {
    // this.titleRef.focus()
    this.setState({
      timestamp: timestampAt(new Date())
    })
    // update the time every 10 seconds
    this.timer = setInterval(()=>this.tick(),10000)
  }

  componentWillUnMount() {
    clearInterval(this.timer)
  }

  tick() {
    let editedTime = this.state.lastChanged
    this.setState({
      lastChanged: editedTime,
      date: timestampAt(new Date())
    })
  }

  handleTitleChange(e) {
    this.setState({
      title: e.target.value,
      titleFocused: true,
      textFocused: false})
  }

  handleTextChange(e) {
    this.setState({
      text: e.target.value,
      titleFocused: false,
      textFocused: true})
  }

  handleSubmit(e) {
    let editedAt = this.state.timestamp.props['time']
    e.preventDefault()
    if (this.state.textFocused){
      this.contentRef.blur()
    }
    if (this.state.titleFocused){
      this.contentRef.focus()
    }
    let newNote = {
      'text': this.contentRef.value,
      'title': this.titleRef.value,
      'id': this.titleRef.value,
      'time': editedAt
    }
    let LS = JSON.parse(
      localStorage.getItem('notes'))
    LS = removeId(LS, this.titleRef.value)
    LS.push(newNote)
    localStorage.setItem('notes',
      JSON.stringify(LS))
    this.setState({ lastChanged: editedAt, saved: true })
  }


  render() {
    return (
      <div className='note'>
        <form className='note-form'
          onSubmit={this.handleSubmit}>
          <div className='group'>
            <input
              type='text'
              required
              name='title'
              ref={ref => this.titleRef = ref}
              placeholder="Title"
              onChange={this.handleTitleChange}
              value={this.state.title}
            />
            <input type='submit' value='submit'/>
            <span className='bar'></span>
          </div>
          <div className='group'>
            <input
              type='text'
              required
              name='content'
              ref={ref => this.contentRef = ref}
              placeholder="Content"
              onChange={this.handleTextChange}
              value={this.state.text}
            />
            <span className='bar'></span>
          </div>
        </form>
        <div id='created-at'>
          {this.state.saved?(
            timestampAt(this.state.lastChanged)
          ):(
            this.state.lastChanged
          )}
        </div>
      </div>
    )
  }
}

Note.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
  titleFocused: PropTypes.bool,
  textFocused: PropTypes.bool,
  lastChanged: PropTypes.string,
  timestamp: PropTypes.string,
}

export default Note
