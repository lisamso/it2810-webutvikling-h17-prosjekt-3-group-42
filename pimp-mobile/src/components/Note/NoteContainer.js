import React, {Component} from 'react'
import { Text, TextInput, TouchableNativeFeedback, View} from "react-native";
import Modal from 'react-native-modal';
import Note from "./Note";
import { styles } from '../../styles.js';

class NoteContainer extends Component {

  constructor(props) {
    super(props);
    this.state = {
      notes: [],
      modalVisible: false,
      title: '',
      text: '',
      count: 0
    };
  }

  componentDidMount() {
    this.newNote('Example', 'Press new note to make a new note');
  }

  toggleModal() {
    this.setState((prevState) => {
      return {modalVisible: !prevState.modalVisible};
    });
  }

  submitNote() {
    if (this.state.title !== '') {
      this.newNote(this.state.title, this.state.text);
    }
    this.setState({title: '', text: ''});
    this.toggleModal()
  }

  removeNote(id) {
    let arr = this.state.notes;
    let index = -1;
    for (let i = 0; i < arr.length; i++) {
      if (arr[i].props.id === id) {
        index = i;
      }
    }
    if (index !== -1) {
      arr.splice(index, 1);
    }
    this.setState({notes: arr});
  }

  newNote(title='',text=''){
    let note =
      <Note
        id={this.state.count}
        title={title}
        text={text}
      />;
    this.setState((prevState) => {
      return {count: prevState.count + 1}
    });
    let arr = this.state.notes;
    arr.push(note);
    this.setState({ notes: arr })
  }

  allNotes() {
    return(
      this.state.notes.map((n,idx) =>
        <View style={styles.row_box} key={idx}>
          <View style={[styles.container, styles.carrot_border]}>
            {n}
          </View>
          <TouchableNativeFeedback
            onPress={() => { this.removeNote(n.props.id)}}
          >
            <View style={styles.delete_container}>
              <View style={[styles.delete_border, styles.carrot_border]}>
                <Text style={[styles.delete_text, styles.carrot]}>x</Text>
              </View>
            </View>
          </TouchableNativeFeedback>
        </View>
      ));
  }

  render(){
    return(
      <View style={styles.box}>
        <View style={[styles.box_header, styles.carrot_border]}>
          <Text style={[styles.title, styles.carrot]}>Your Notes</Text>
        </View>
        <View style={[styles.box_body, styles.carrot_border]}>
          {this.allNotes()}
        </View>
        <TouchableNativeFeedback
          onPress={() => { this.toggleModal()}}>
          <View style={[styles.box_button, styles.carrot_background]}>
            <Text style={styles.wet_asphalt}>
              New note
            </Text>
          </View>
        </TouchableNativeFeedback>

        <Modal
          animationIn="slideInLeft"
          animationOut="slideOutRight"
          transparent={true}
          isVisible={this.state.modalVisible}
          backDropColor="white"
          onBackdropPress={() => {this.toggleModal()}}
          onBackButtonPress={() => {this.toggleModal()}}
        >
          <View style={styles.box}>
            <View style={[styles.box_header, styles.carrot_border]}>
              <Text style={[styles.title, styles.carrot]}>New note</Text>
            </View>
            <View style={[styles.box_body, styles.carrot_border]}>
              <TextInput
                style={{height: 60, color: '#e67e22'}}
                placeholder="Title"
                onChangeText={(text) => this.setState({title: text})}
              />
              <TextInput
                style={{height: 60, color: '#e67e22'}}
                placeholder="Text"
                onChangeText={(text) => this.setState({text: text})}
              />
            </View>
            <TouchableNativeFeedback
              onPress={() => {this.submitNote()}}>
              <View style={[styles.box_button, styles.carrot_background]}>
                <Text style={styles.wet_asphalt}>
                    Confirm
                </Text>
              </View>
            </TouchableNativeFeedback>
          </View>
        </Modal>
      </View>

    )
  }
}

NoteContainer.propTypes = {

}


export default NoteContainer

