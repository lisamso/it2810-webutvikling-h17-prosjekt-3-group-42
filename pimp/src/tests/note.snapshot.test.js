import Note from '../components/Note/Note';
import NoteContainer from '../components/Note/NoteContainer';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';


configure({ adapter: new Adapter() });



test('Note component should render as expected', () =>{
  const component = shallow(<Note />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

describe('Test the functions of the Note component.', () =>{

  const note = {title: "Tittel", text: "Tekst"};

  it('Checks if changes to the input field named title changes the title state value of the Note component.', () =>{
    const component = shallow(<Note note={note} />);
    const title = component.find('#tittelid');
    title.simulate('change', { target: { value: 'Hello' } });
    expect(component.state().title).toBe('Hello');
  });

  it('Checks if changes to the input field named text changes the text state value of the Note component.', () =>{
    const component = shallow(<Note note={note} />);
    const text = component.find('#tekstid');
    text.simulate('change', { target: { value: 'Hello' } });
    expect(component.state().text).toBe('Hello');
  });

  it('', () =>{
    NoteContainer.newNote('tittel', 'tekst');
    
  });

});
