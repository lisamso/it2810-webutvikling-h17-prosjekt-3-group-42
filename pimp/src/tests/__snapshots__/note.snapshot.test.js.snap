// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component should render as expected 1`] = `
<div
  className="note"
>
  <form
    className="note-form"
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <input
        id="tittelid"
        name="title"
        onChange={[Function]}
        placeholder="Title"
        required={true}
        type="text"
      />
      <input
        id="submitid"
        type="submit"
        value="submit"
      />
      <span
        className="bar"
      />
    </div>
    <div
      className="group"
    >
      <input
        id="tekstid"
        name="content"
        onChange={[Function]}
        placeholder="Content"
        required={true}
        type="text"
      />
      <span
        className="bar"
      />
    </div>
  </form>
  <div
    id="created-at"
  />
</div>
`;
